### ISSUE TITLE
Refactor Test Matrix

### ISSUE THREAD

**Issue by insightindustry** [insightindustry/sqlathanor#109]
Currently, the SQLAthanor test matrix is ungainly and badly designed. This is partially because of the design decision to support Python 2.7 and to support older versions of SQLAlchemy and other dependencies. However, the test matrix as a whole needs to be refactored because with the newer versions of Python and newer versions of dependencies, full dependency expansion in the test matrix blows up the number of jobs beyond 200.

Which Travis CI does not support.

Bottom line, the test matrix needs to be refactored to find a method to maintain tests across different dependency versions while still keeping the number of jobs to a minimum.
